### ISSUE TITLE
Compact Spinbox does not look good on all platforms

### ISSUE THREAD

**Issue by polwel** [pyqtgraph/pyqtgraph#701]
Pull request #410 introduced a `compactHeight` option for `Spinbox` that is on by default. It reduces the height of the SpinBox, as otherwise it would be overly large on many platforms. The new system does not look good either.

For instance on Windows 7, it crushes the buttons to step the value. Also, in a ParameterTree, the Spinbox does not even fill the row.

**After PR #410:**

![screenshot_1](https://user-images.githubusercontent.com/7679808/40838999-dfe3a8e4-65a0-11e8-86ed-6b900507373a.png)

![screenshot_2](https://user-images.githubusercontent.com/7679808/40839000-dfff63cc-65a0-11e8-9a80-8bb65d45bdd1.png)

**Before PR #410:**

![screenshot_2](https://user-images.githubusercontent.com/7679808/40839047-11d63c04-65a1-11e8-9572-2e3ddb060f09.png)

![screenshot_1](https://user-images.githubusercontent.com/7679808/40839046-11b44914-65a1-11e8-9bde-ccdf5bf50f3d.png)

Clearly a better solution is required.

**Comment by campagnola** [comment]
Agreed, this has irritated me for years. I am working on getting windows and osx CI implemented, so if we can devise a way to unit test this, then there might be some chance of getting consistency across platforms.

**Comment by jaxankey** [comment]
Since version 0.11.0, I have noticed the SpinBox is crushed as well. Here is a minimal script:

```
# Set up the QApp
import pyqtgraph
_qtapp = pyqtgraph.QtCore.QCoreApplication.instance()
if not _qtapp: _qtapp = pyqtgraph.QtGui.QApplication([])

import pyqtgraph

# Create the various objects.
w = pyqtgraph.QtGui.QMainWindow()
g = pyqtgraph.QtGui.QWidget()
l = pyqtgraph.QtGui.QGridLayout()
s = pyqtgraph.SpinBox()

# Set the layout and central widget of the window.
g.setLayout(l)
w.setCentralWidget(g)

# Add the spin box to the layout.
l.addWidget(s)

# Show the window.
w.show()
```
The output is a crushed SpinBox:

![image](https://user-images.githubusercontent.com/8275367/87556503-bb628480-c684-11ea-989c-5e4eb0af529d.png)

If I include `s.setOpts(compactHeight=False)`, I get this, which is also pretty bananas:

![image](https://user-images.githubusercontent.com/8275367/87556691-f2389a80-c684-11ea-881c-7358e62ce204.png)

Hope this can be fixed / updated in conda soon. I'm deploying to students in the Fall and this visual blip will chase them away from the otherwise beautiful world of pyqtgraph.

Cheers,
Jack

PyQtGraph version: v0.11.0
Qt Python binding: 'PyQt5 5.9.2 Qt 5.9.6'
Python version: 3.7
NumPy version: 1.18.5
Operating system: Linux Mint 20
Installation method: conda
